server/integration/testsuite/src/test/java/org/jgroups/protocols/PARTITION_HANDLING.java

```diff
@@ -0,0 +1,155 @@
+package org.jgroups.protocols;
+
+import org.infinispan.server.test.util.PartitionHandlingController;
+import org.jgroups.Global;
+import org.jgroups.Header;
+import org.jgroups.Message;
+import org.jgroups.annotations.MBean;
+import org.jgroups.conf.ClassConfigurator;
+import org.jgroups.logging.Log;
+import org.jgroups.logging.LogFactory;
+import org.jgroups.stack.Protocol;
+import org.jgroups.util.Streamable;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Supplier;
+
+/**
+ * This is custom protocol used for partition handling. What it basically does is, that it discards specific
+ * JGroups messages from some allowed nodes. Do not use this class directly (via JGroups Probe), use {@link PartitionHandlingController} instead.
+ * <p>
+ * The class file is inserted as a JAR into the test servers (see pom.xml).
+ *
+ * @author Jiri Holusa (jholusa@redhat.com)
+ * @author Radim Vansa (rvansa@redhat.com)
+ */
+@MBean(description = "Enables custom partition handling.")
+public class PARTITION_HANDLING extends Protocol {
```

Comment:
I wonder if you could do the same thing using `DISCARD.addIgnoredMember()`/`resetIgnoredMembers()`. I have no idea if they work over JMX/probe, but I'd like to avoid the extra jar if possible.

Reply:
I guess I have the same question: could this be achieved by DISCARD and if not, why not? If this is generally useful it should ideally reside in JGroups repo which solves the "extra jar" problem, but mainly this way it can be reused in other projects like WildFly.

Reply:
Looking at DISCARD it seems it could be used, yes. I'll try to use it and see if there are any issues.

Reply:
So I found the problem - with probe, you can only call methods that have primitive types as arguments, and the DISCARD.addIgnoreMember method expects an argument of type Address. So DISCARD could be used, but I'd need to make changes to it and we'd need to wait for a JGroups release with them.

Reply:
Or we could use DISCARD only on the node that is alone in a partition, but that would limit the possible splits on clusters of size 4 and more (you wouldn't be able to do 2 partitions of 2 nodes). Since we only test with max 3 servers this would work.

Reply:
> with probe, you can only call methods that have primitive types as arguments

Nice find; JGroups Jira + PR would be nice.

>  Since we only test with max 3 servers this would work.

Looks like this could work to. Would you mind giving it a try? I would be curious to see how fast you can get the partitions to merge then.